components/automate-ui/src/app/services/node-details/node-norunid-resolver.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { Router, Resolve, RouterStateSnapshot,
+         ActivatedRouteSnapshot } from '@angular/router';
+
+import { NodeRun } from '../../types/types';
+import { NodeRunsService } from './node-runs.service';
+
+@Injectable()
+export class NodeNoRunIdResolverService implements Resolve<NodeRun> {
+  constructor(private service: NodeRunsService, private router: Router) {}
+
+  resolve(route: ActivatedRouteSnapshot, _state: RouterStateSnapshot): Promise<NodeRun> {
+    const nodeId = route.paramMap.get('node-id');
+
+    return this.service.getNodeRunsByID( nodeId ).
+      then(nodeRuns => {
+        // if the node runs is found we select the latest run-id.
+        // Go to node runs detail page
+        if (nodeRuns && Object.keys(nodeRuns[0]).length !== 0) {
```

Comment:
Once I annotated `getNodeRunsByID` with the correct type, I uncovered some issues here. I think you are trying to check if the first node run is not a null object.  Should first be checking that a first node run even exists--that's the `nodeRuns.length` check here. Then, we could check if `nodeRuns[0] !== null` but I think it is more clear to just directly check for what we care about, the `runId`. So we end up with this:
```suggestion
        if (nodeRuns && nodeRuns.length && nodeRuns[0].runId) {
```
But can we really get a nodeRuns array returned with null objects in it? I have no idea on that point, but if not it could be simplified to:
```
        if (nodeRuns && nodeRuns.length) {
```